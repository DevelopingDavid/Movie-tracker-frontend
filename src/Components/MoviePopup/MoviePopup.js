import React from 'react';

const MoviePopup = (props) => {
  return (
    <div>
      {props.foundMovie.title}
      <img src={`https://image.tmdb.org/t/p/w500/${props.foundMovie.img}`}/>
    </div>
  )
}

export default MoviePopup;