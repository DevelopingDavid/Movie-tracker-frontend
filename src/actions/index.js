export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies,
});

export const loginUser = (loggedIn) => ({
  type: 'LOGIN_USER',
  loggedIn
});