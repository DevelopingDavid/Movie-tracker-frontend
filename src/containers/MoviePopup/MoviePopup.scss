.popup-container {
  display: flex;
  font-family: 'Open Sans', sans-serif;
}
.poster-container {
  margin: 30px;
  img {
    width: 400px;
  }
}

.info-container {
  margin: 14px;
  display: flex;
  color: white;
  flex-direction: column;
  align-items: flex-start;
  div {
    display: flex;
    align-items: center;
    i {
      margin-left: 30px;
      margin-top: 7px;
      font-size: 2rem;
      color: white;
      &:hover {
        color:#940002;
      }
    }
  }
  h1 {
    font-size: 3rem;
  }
  p {
    text-align: left;
    margin: 10px 0;
    font-size: 2rem;
  }
  a {
    color: white;
    & :visited {
      color: white;
    }
  }
}