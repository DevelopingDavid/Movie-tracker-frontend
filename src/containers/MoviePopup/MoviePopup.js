import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setFavoriteMovies, loginUser } from '../../actions';
import { Link, Redirect} from 'react-router-dom';


export class MoviePopup extends Component {

  addFavoriteToDatabase = async () => {
    const movieInfo = this.props.foundMovie;
    const newMovieInfo = { 
      movie_id: movieInfo.movie_id,
      user_id: this.props.user.id, 
      title: movieInfo.title, 
      poster_path: movieInfo.poster_path, 
      release_date: movieInfo.release_date, 
      vote_average: movieInfo.vote_average, 
      overview: movieInfo.overview
    };
    const options = {
      method: 'post',
      headers: {'Content-Type': 'application/json'},
      body: JSON.stringify(newMovieInfo)
    }
    const favoriteDatabase = 'http://localhost:3000/api/users/favorites/new';
    const response = await fetch(favoriteDatabase, options);
    if (response.ok) {
      const favoriteMovies = await response.json();
      return favoriteMovies;
    } else {
      return `can't fetch movies`;
    }
  }

  deleteFavorites = async () => {
    const { user, foundMovie } = this.props
    const deleteURL = `http://localhost:3000/api/users/${user.id}/favorites/${foundMovie.movie_id}`;
    const options = {
      method: 'DELETE',
      body: [user.id, foundMovie.movie_id]
    }
    const response = await fetch(deleteURL, options);
    const data = await response.json();
    console.log(data)
  }

  setFavorites = () => {
    let { foundMovie, setFavoriteMovies } = this.props
    setFavoriteMovies(foundMovie.movie_id);
    this.addFavoriteToDatabase();
  }

  render () {
    let { foundMovie } = this.props
    if(Object.keys(this.props.user).length === 0) { 
      let user = JSON.parse(localStorage.getItem('movieTrackerUser'));
      if (user) {
        this.props.loginUser(user.id, user.name);
        return <Redirect to='/movies'/>;
      } else {
        return <Redirect to='/sign-in'/>;
      }
    }
    const background = {
      backgroundImage: `url(http://image.tmdb.org/t/p/original${foundMovie.backdrop_path})`,
      backgroundColor: `rgba(5, 5, 5, 0.815)`,
      backgroundBlendMode: `multiply`,
      backgroundRepeat: `no-repeat`,
      backgroundSize: `cover`
    };
  
    return (
      <section className="popup-container" style={background}>
        <div className="poster-container">
          <img alt='movie poster' src={`https://image.tmdb.org/t/p/w500/${foundMovie.poster_path}`}/>
        </div>
        <div className="info-container">
          <div>
            <h1>{foundMovie.title}</h1>
            <i id='heart-favorite' className="fas fa-heart" onClick={this.setFavorites}></i>
            <i onClick={this.deleteFavorites} className="fa fa-trash" aria-hidden="true"></i>
            <Link to='/movies'><i class="fas fa-undo"></i></Link>
          </div>
          <h1>{`(${foundMovie.release_date})`}</h1>
          <p>{foundMovie.overview}</p>
        </div>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  setFavoriteMovies: (id) => dispatch(setFavoriteMovies(id)),
  loginUser: (id, name) => dispatch(loginUser(id, name))
});

export default connect(mapStateToProps, mapDispatchToProps)(MoviePopup)