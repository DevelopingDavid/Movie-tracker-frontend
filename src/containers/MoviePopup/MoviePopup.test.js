import React from 'react';
import { MoviePopup, mapDispatchToProps, mapStateToProps} from '../MoviePopup/MoviePopup';
import { shallow } from 'enzyme';
import { setFavoriteMovies, loginUser } from '../../actions';

describe('MoviePopup', () => {
  let wrapper;
  let mockState = {
    user: {
      id: 76,
      name: 'matt'
    }
  }

  let mockFoundMovie = {}

  beforeEach(() => {
    wrapper = shallow( 
      <MoviePopup user={mockState}
                  foundMovie={mockFoundMovie}/>
    ) 
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a user object', () => {
      const predicted = {
        user: {
          id: 76,
          name: 'matt'
      }}
      
      const mappedProps = mapStateToProps(mockState);
      
      expect(mappedProps).toEqual(predicted)
    });
  });

  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch when using mapDispatchToProps', () => {
  //     const mockDispatch = jest.fn();
  //     const actionToDispatch = setFavoriteMovies(1);

  //     const mappedProps = mapDispatchToProps(mockDispatch);
      
  //     mappedProps.setFavoriteMovies(1);

  //     expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  //   });
  // });
});