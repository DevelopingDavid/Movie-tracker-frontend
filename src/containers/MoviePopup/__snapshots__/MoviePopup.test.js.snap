// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePopup should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MoviePopup
    foundMovie={Object {}}
    loginUser={[MockFunction]}
    setFavoriteMovies={Object {}}
    user={
      Object {
        "user": Object {
          "id": 76,
          "name": "matt",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="poster-container"
        >
          <img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w500/undefined"
          />
        </div>,
        <div
          className="info-container"
        >
          <div>
            <h1 />
            <i
              className="fas fa-heart"
              id="heart-favorite"
              onClick={[Function]}
            />
            <i
              aria-hidden="true"
              className="fa fa-trash"
              onClick={[Function]}
            />
            <Link
              to="/movies"
            >
              <i
                className="fas fa-undo"
              />
            </Link>
          </div>
          <h1>
            (undefined)
          </h1>
          <p />
        </div>,
      ],
      "className": "popup-container",
      "style": Object {
        "backgroundBlendMode": "multiply",
        "backgroundColor": "rgba(5, 5, 5, 0.815)",
        "backgroundImage": "url(http://image.tmdb.org/t/p/originalundefined)",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="movie poster"
            src="https://image.tmdb.org/t/p/w500/undefined"
          />,
          "className": "poster-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "src": "https://image.tmdb.org/t/p/w500/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h1 />
              <i
                className="fas fa-heart"
                id="heart-favorite"
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="fa fa-trash"
                onClick={[Function]}
              />
              <Link
                to="/movies"
              >
                <i
                  className="fas fa-undo"
                />
              </Link>
            </div>,
            <h1>
              (undefined)
            </h1>,
            <p />,
          ],
          "className": "info-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <i
                  className="fas fa-heart"
                  id="heart-favorite"
                  onClick={[Function]}
                />,
                <i
                  aria-hidden="true"
                  className="fa fa-trash"
                  onClick={[Function]}
                />,
                <Link
                  to="/movies"
                >
                  <i
                    className="fas fa-undo"
                  />
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-heart",
                  "id": "heart-favorite",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-trash",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <i
                    className="fas fa-undo"
                  />,
                  "to": "/movies",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-undo",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(undefined)",
            },
            "ref": null,
            "rendered": "(undefined)",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="poster-container"
          >
            <img
              alt="movie poster"
              src="https://image.tmdb.org/t/p/w500/undefined"
            />
          </div>,
          <div
            className="info-container"
          >
            <div>
              <h1 />
              <i
                className="fas fa-heart"
                id="heart-favorite"
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="fa fa-trash"
                onClick={[Function]}
              />
              <Link
                to="/movies"
              >
                <i
                  className="fas fa-undo"
                />
              </Link>
            </div>
            <h1>
              (undefined)
            </h1>
            <p />
          </div>,
        ],
        "className": "popup-container",
        "style": Object {
          "backgroundBlendMode": "multiply",
          "backgroundColor": "rgba(5, 5, 5, 0.815)",
          "backgroundImage": "url(http://image.tmdb.org/t/p/originalundefined)",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="movie poster"
              src="https://image.tmdb.org/t/p/w500/undefined"
            />,
            "className": "poster-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie poster",
              "src": "https://image.tmdb.org/t/p/w500/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h1 />
                <i
                  className="fas fa-heart"
                  id="heart-favorite"
                  onClick={[Function]}
                />
                <i
                  aria-hidden="true"
                  className="fa fa-trash"
                  onClick={[Function]}
                />
                <Link
                  to="/movies"
                >
                  <i
                    className="fas fa-undo"
                  />
                </Link>
              </div>,
              <h1>
                (undefined)
              </h1>,
              <p />,
            ],
            "className": "info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <i
                    className="fas fa-heart"
                    id="heart-favorite"
                    onClick={[Function]}
                  />,
                  <i
                    aria-hidden="true"
                    className="fa fa-trash"
                    onClick={[Function]}
                  />,
                  <Link
                    to="/movies"
                  >
                    <i
                      className="fas fa-undo"
                    />
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-heart",
                    "id": "heart-favorite",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-trash",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <i
                      className="fas fa-undo"
                    />,
                    "to": "/movies",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-undo",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(undefined)",
              },
              "ref": null,
              "rendered": "(undefined)",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
